application/src/main/java/md/maib/retail/application/campaigns_list_by_date/CampaignsListByDateService.java

```diff
@@ -16,11 +19,17 @@ public CampaignsListByDateService(Campaigns campaigns) {
 
     @Override
     public List<CampaignSomeInfo> activeCampaignsByDate(LocalDate date) {
-
         return campaigns.listByDate(date)
                 .stream()
+                .filter(campaign -> isDateWithinInterval(CampaignSomeInfo.valueOf(campaign), date) && campaign.getState() == CampaignState.ACTIVE)
```

Comment:
filter for date within interval must be realized in adapter/persistence listByDate method